zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/MetricFromProcessInstanceHandler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.handlers;
+
+import static io.camunda.exporter.utils.ExporterUtil.tenantOrDefault;
+import static io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent.ELEMENT_ACTIVATING;
+
+import io.camunda.exporter.store.BatchRequest;
+import io.camunda.webapps.schema.entities.operate.MetricEntity;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceRecordValue;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+public class MetricFromProcessInstanceHandler
+    implements ExportHandler<MetricEntity, ProcessInstanceRecordValue> {
+  protected static final int EMPTY_PARENT_PROCESS_INSTANCE_ID = -1;
+  protected static final String EVENT_PROCESS_INSTANCE_STARTED = "EVENT_PROCESS_INSTANCE_STARTED";
+
+  private final String indexName;
+
+  public MetricFromProcessInstanceHandler(final String indexName) {
+    this.indexName = indexName;
+  }
+
+  @Override
+  public ValueType getHandledValueType() {
+    return ValueType.PROCESS_INSTANCE;
+  }
+
+  @Override
+  public Class<MetricEntity> getEntityType() {
+    return MetricEntity.class;
+  }
+
+  @Override
+  public boolean handlesRecord(final Record<ProcessInstanceRecordValue> record) {
+    final var recordValue = record.getValue();
+    final boolean isRootProcessInstance =
+        recordValue.getParentProcessInstanceKey() == EMPTY_PARENT_PROCESS_INSTANCE_ID;
+    return isRootProcessInstance && record.getIntent().equals(ELEMENT_ACTIVATING);
+  }
+
+  @Override
+  public List<String> generateIds(final Record<ProcessInstanceRecordValue> record) {
+    return List.of();
+  }
+
+  @Override
+  public MetricEntity createNewEntity(final String id) {
+    return new MetricEntity();
+  }
```

Comment:
PS: these two functions (not generating Id, and not setting it on entity) were like this on https://github.com/camunda/camunda/blob/feat-operate-exporter/zeebe/exporters/operate-exporter/src/main/java/io/camunda/zeebe/operate/exporter/handlers/MetricFromProcessInstanceHandler.java.

I assumed this was intentional, right?

Reply:
Since this is a metric index probably ES/OS create random document IDs for them? cc @sdorokhova maybe you know better regarding this

Reply:
I had a discussion with @sdorokhova about this.
If ids were not generated the entities won't be updated or cached, and hence not picked up for flushing ([https://github.com/camunda/camunda/blob/e78dd201e46fbd3897e44b64bd78b7693916dd46/z[…]rc/main/java/io/camunda/exporter/store/ExporterBatchWriter.java](https://github.com/camunda/camunda/blob/e78dd201e46fbd3897e44b64bd78b7693916dd46/zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ExporterBatchWriter.java#L37))

We agreed to use processInstanceKey as an id here.